NPE when opening stack of crafting cards

ava.lang.NullPointerException: Rendering screen
	at logisticspipes.gui.GuiCraftingPipe.func_146979_b(GuiCraftingPipe.java:318)
	at net.minecraft.client.gui.inventory.GuiContainer.func_73863_a(GuiContainer.java:117)
	at logisticspipes.utils.gui.LogisticsBaseGuiScreen.func_73863_a(LogisticsBaseGuiScreen.java:202)
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:396)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1124)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1119)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:398)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

[13:26:45] [pool-2-thread-1/INFO] [minecraft/GuiNewChat]: [CHAT] You are running Logistics Pipes from a development environment.
[13:26:45] [Server thread/INFO] [minecraft/IntegratedServer]: Saving and pausing game...
[13:26:45] [Server thread/INFO] [minecraft/MinecraftServer]: Saving chunks for level 'New World'/overworld
[13:26:45] [main/FATAL] [minecraft/Minecraft]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_211]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_211]
	at net.minecraft.util.Util.runTask(Util.java:54) [Util.class:?]
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1177) [Minecraft.class:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:441) [Minecraft.class:?]
	at net.minecraft.client.main.Main.main(Main.java:118) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_211]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_211]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_211]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_211]
	at net.minecraftforge.gradle.GradleStartCommon.launch(GradleStartCommon.java:97) [start/:?]
	at GradleStart.main(GradleStart.java:25) [start/:?]
Caused by: java.lang.NullPointerException
	at logisticspipes.network.packets.BufferTransfer.processPacket(BufferTransfer.java:34) ~[BufferTransfer.class:?]
	at logisticspipes.network.PacketInboundHandler.inThreadProcessPacket(PacketInboundHandler.java:36) ~[PacketInboundHandler.class:?]
	at logisticspipes.network.PacketInboundHandler.lambda$channelRead0$0(PacketInboundHandler.java:30) ~[PacketInboundHandler.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_211]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266) ~[?:1.8.0_211]
	at java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:1.8.0_211]
	at net.minecraft.util.Util.runTask(Util.java:53) ~[Util.class:?]
	... 15 more
[13:26:46] [Thread-22/INFO] [THAUMCRAFT]: Stopping aura thread for dim 1
[13:26:46] [Thread-21/INFO] [THAUMCRAFT]: Stopping aura thread for dim -1
[13:26:46] [Thread-25/INFO] [minecraft/GuiNewChat]: [CHAT] WARNING: BuildCraft 7.99.19 is in ALPHA!
[13:26:46] [Thread-25/INFO] [minecraft/GuiNewChat]: [CHAT]   Report BuildCraft bugs you find here
[13:26:46] [Thread-25/INFO] [minecraft/GuiNewChat]: [CHAT]   and include the BuildCraft version 7.99.19 in the description
[13:26:46] [Thread-25/INFO] [minecraft/GuiNewChat]: [CHAT]   If you have performance problems then try disabling
[13:26:46] [Thread-25/INFO] [minecraft/GuiNewChat]: [CHAT] everything in the BuildCraft perfomance config section.
[13:27:15] [Server thread/INFO] [minecraft/IntegratedServer]: Saving and pausing game...
[13:27:15] [Server thread/INFO] [minecraft/MinecraftServer]: Saving chunks for level 'New World'/overworld
[13:27:17] [main/INFO] [jei]: Updating ingredient filter...
[13:27:17] [main/INFO] [jei]: Updated  ingredient filter in 0 ms
[13:27:20] [main/INFO] [minecraft/GuiNewChat]: [CHAT] You are now playing creative+
[13:27:21] [main/INFO] [minecraft/GuiNewChat]: [CHAT] You are now playing adventure
[13:28:22] [Server thread/WARN] [minecraft/MinecraftServer]: Can't keep up! Did the system time change, or is the server overloaded? Running 56450ms behind, skipping 1129 tick(s)
[13:30:17] [Server thread/WARN] [minecraft/MinecraftServer]: Can't keep up! Did the system time change, or is the server overloaded? Running 114276ms behind, skipping 2285 tick(s)
[13:30:21] [Server thread/WARN] [minecraft/MinecraftServer]: Can't keep up! Did the system time change, or is the server overloaded? Running 4780ms behind, skipping 95 tick(s)
[13:30:36] [Server thread/INFO] [minecraft/MinecraftServer]: Stopping server
[13:30:36] [Server thread/INFO] [minecraft/MinecraftServer]: Saving players
[13:30:36] [Server thread/INFO] [minecraft/MinecraftServer]: Saving worlds
[13:30:36] [Server thread/INFO] [minecraft/MinecraftServer]: Saving chunks for level 'New World'/overworld
[13:30:36] [Server thread/INFO] [THAUMCRAFT]: Removing aura cache for world 0
[13:30:37] [Thread-20/WARN] [THAUMCRAFT]: No auras found!
[13:30:37] [Thread-20/INFO] [THAUMCRAFT]: Stopping aura thread for dim 0
[13:30:41] [Server thread/INFO] [FML]: Unloading dimension 0
[13:30:43] [Server thread/INFO] [FML]: Applying holder lookups
[13:30:43] [Server thread/INFO] [FML]: Holder lookups applied
[13:30:45] [main/FATAL] [minecraft/Minecraft]: Reported exception thrown!
net.minecraft.util.ReportedException: Rendering screen
	at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1204) ~[EntityRenderer.class:?]
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1208) ~[Minecraft.class:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:441) [Minecraft.class:?]
	at net.minecraft.client.main.Main.main(Main.java:118) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_211]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_211]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_211]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_211]
	at net.minecraftforge.gradle.GradleStartCommon.launch(GradleStartCommon.java:97) [start/:?]
	at GradleStart.main(GradleStart.java:25) [start/:?]
Caused by: java.lang.NullPointerException
	at logisticspipes.gui.GuiCraftingPipe.drawGuiContainerForegroundLayer(GuiCraftingPipe.java:318) ~[GuiCraftingPipe.class:?]
	at net.minecraft.client.gui.inventory.GuiContainer.drawScreen(GuiContainer.java:135) ~[GuiContainer.class:?]
	at logisticspipes.utils.gui.LogisticsBaseGuiScreen.drawScreen(LogisticsBaseGuiScreen.java:202) ~[LogisticsBaseGuiScreen.class:?]
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:381) ~[ForgeHooksClient.class:?]
	at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1177) ~[EntityRenderer.class:?]
	... 15 more
[13:30:45] [main/INFO] [STDOUT]: [net.minecraft.init.Bootstrap:printToSYSOUT:629]: ---- Minecraft Crash Report ----

WARNING: coremods are present:
  ForgelinPlugin (Forgelin-1.8.3.jar)
  LogisticsPipesCoreLoader (LogisticsPipes_dummy.jar)
Contact their authors BEFORE contacting forge

// I'm sorry, Dave.

Time: 6/22/19 1:30 PM
Description: Rendering screen

stacktraces from debugger

--- a/common/logisticspipes/modules/ModuleCrafter.java
+++ b/common/logisticspipes/modules/ModuleCrafter.java
@@ -11,6 +11,8 @@ import java.util.UUID;
 import java.util.concurrent.DelayQueue;
 import java.util.stream.Collectors;
 
+import javax.annotation.Nonnull;
+
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.entity.EntityPlayerSP;
 import net.minecraft.entity.player.EntityPlayer;
@@ -1472,9 +1474,9 @@ public class ModuleCrafter extends LogisticsGuiModule implements ICraftItems, IH
 	}
 
 	public static class ClientSideSatelliteNames {
-		public String satelliteName;
-		public String[] advancedSatelliteNameArray;
-		public String liquidSatelliteName;
-		public String[] liquidSatelliteNameArray;
+		public @Nonnull String satelliteName = "";
+		public @Nonnull String[] advancedSatelliteNameArray = {};
+		public @Nonnull String liquidSatelliteName = "";
+		public @Nonnull String[] liquidSatelliteNameArray = {};
 	}
 }

